lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Context.java

```diff
@@ -84,17 +87,27 @@ public void applyCredentialProviderOverride(AwsRequestOverrideConfiguration.Buil
         credentialsProviderOverride.ifPresent(builder::credentialsProvider);
     }
 
-    record S3SseContext(S3FileSystemConfig.S3SseType sseType, String sseKmsKeyId)
+    record S3SseContext(S3SseType sseType, String sseKmsKeyId, S3SseCustomerKey sseCustomerKey)
```

Comment:
```suggestion
record S3SseContext(S3SseType sseType, String sseKmsKeyId, Optional<S3SseCustomerKey> sseCustomerKey)
```

Optional feels (naively) more appropriate to me than null.

Reply:
done